classes/Hook.php

```diff
@@ -885,7 +920,48 @@ public static function exec(
             }
         }
 
-        return $output;
+        if ($array_return) {
+            return $output;
+        } else {
+            return ($live_edit ? '<div id="'.$hook_name.'" class="dndHook" style="min-height:50px">' : '').$output.($live_edit ? '</div>' : '');
+        }// Return html string
+    }
+
+    /**
+     * Regenerate content of hook by adding html identifier
+     *
+     * @param string $display Get content for this hook
+     * @param string $moduleInstance Get modules content
+     * @param int $id_hook Get hook id
+     * @return string
+     */
+    public static function wrapLiveEdit($display, $moduleInstance, $id_hook)
+    {
+
+        $context = Context::getContext();
+
+        $link_module = $context->link->getBaseLink() . Tools::getValue('ad') . '/index.php?controller=AdminModules&configure=' . Tools::safeOutput($moduleInstance->name) . '&token=' . Tools::getAdminToken('AdminModules'.(int)Tab::getIdFromClassName('AdminModules').(int)Tools::getValue('id_employee'));
+
+        $is_configurable = method_exists($moduleInstance, 'getContent') ? 1 : 0;
+
+        $live_content = '<div rel="'.Tools::safeOutput($moduleInstance->name).'" id="hook_'.(int)$id_hook.'_module_'.(int)$moduleInstance->id.'_moduleName_'.str_replace('_', '-', Tools::safeOutput($moduleInstance->name)).'"
+				class="dndModule">
+					<div class="toolbar">
+					    <div class="toolbar-content">
+						<img src="'._MODULE_DIR_.Tools::safeOutput($moduleInstance->name).'/logo.png">
+                        <div class="toolbar-title">' . Tools::safeOutput($moduleInstance->displayName) . '</div></div><div class="toolbar-btn">
+				<a href="#" id="'.(int)$id_hook.'_'.(int)$moduleInstance->id.'" class="moveModule">
+					<i class="material-icons md-icon">zoom_out_map</i></a>
+				<a href="#" id="'.(int)$id_hook.'_'.(int)$moduleInstance->id.'" class="unregisterHook">
+					<i class="material-icons md-icon">delete_forever</i></a> ';
+        if (1 === $is_configurable) {
+            $live_content .= '<a target="_blank" href="'.$link_module.'" id="'.(int)$id_hook.'_'.(int)$moduleInstance->id.'" class="settingModule">
+					<i class="material-icons md-icon">settings</i></a>';
+        }
+
+        $live_content .= '</div></div>'.$display.'</div>';
```

Comment:
Why not use `$this->context->smarty->createTemplate()` to generate wrapped `$live_content` html ?

Reply:
Hey,
In this method, i just inject content in the tpl. So i dont need to create a new template.
Thanks